### ISSUE TITLE
NotFoundError

### ISSUE THREAD

**Issue by LJXLJXLJX** [MaybeShewill-CV/CRNN_Tensorflow#164]
I try to run the demo for one image, but a notfounderror ocurr .
What should I do?

```
NotFoundError (see above for traceback): Restoring from checkpoint failed. This is most likely due to a Variable name or other graph key that is missing from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:

Key shadow/batch_normalization/beta not found in checkpoint
	 [[Node: save/RestoreV2 = RestoreV2[dtypes=[DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, ..., DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT], _device="/job:localhost/replica:0/task:0/device:CPU:0"](_arg_save/Const_0_0, save/RestoreV2/tensor_names, save/RestoreV2/shape_and_slices)]]
	 [[Node: save/RestoreV2/_3 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device_incarnation=1, tensor_name="edge_7_save/RestoreV2", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]
```

**Comment by MaybeShewill-CV** [comment]
@LJXLJXLJX May be the wrong path of the model weights path

**Comment by LJXLJXLJX** [comment]
@MaybeShewill-CV I don't think so, the wrong path will lead to another error but no this one.

**Comment by MaybeShewill-CV** [comment]
@LJXLJXLJX You can convert it into frozen pb file and check if the node name in the graph are not compareable with the one written in the src code

**Comment by sundarkrish** [comment]
I too get the same error.Since I am a beginner ,please @MaybeShewill-CV  can you help me solve this issue ?

**Comment by mdbenito** [comment]
I think this is due to the recent upgrade to TF 1.10. The batch normalization layers are now using `tf.layers.batch_normalization`, instead of `tf.contrib.layers.batch_norm`. This means that the checkpoint in the repo is no longer valid for the current code! I hadn't thought of this as I submitted the PR. Maybe we should undo that change (and any having the same consequence)?

(And... that's why one always wants to have regression tests! ;)

**Comment by MaybeShewill-CV** [comment]
@mdbenito I will retrained a new model based on the new code and there will be not such problem at that time I think.

**Comment by MaybeShewill-CV** [comment]
@MarStarck Do not have enough spare time for this now==!. You guys can also retrain a new model based on the Synth90K dataset:)

**Comment by MarStarck** [comment]
@MaybeShewill-CV How big is your train_feature.tfrecord? I'm downloading Synth90K dataset and it is really slow..

**Comment by MaybeShewill-CV** [comment]
@MarStarck I remember the whole dataset is quite big I use part of that to train the model but I can't tell you how big the file is since that data was no longer on my computer now==!

**Comment by yongcong1415** [comment]
my tensorflow 1.11.0 ,the same error to,i need to change tensorflow at 1.10?

**Comment by MaybeShewill-CV** [comment]
@yongcong1415 The old model weights are trained with tensorflow 1.3 but the code here has been upgraded to tensorflow 1.10. Maybe the fastest way is to train a new model under the new code:)

**Comment by Spatialghost** [comment]
Hi everyone!
I trained model at my own dataset and i have this error:
InvalidArgumentError (see above for traceback): Restoring from checkpoint failed. This is most likely due to a mismatch between the current graph and the graph from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:

Assign requires shapes of both tensors to match. lhs shape= [512,37] rhs shape= [512,568]
         [[Node: save/Assign_8 = Assign[T=DT_FLOAT, _class=["loc:@shadow/LSTMLayers/w"], use_locking=true, validate_shape=true, _device="/job:localhost/replica:0/task:0/device:GPU:0"](shadow/LSTMLayers/w, save/RestoreV2/_17)]]
         [[Node: save/RestoreV2/_30 = _Send[T=DT_FLOAT, client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device_incarnation=1, tensor_name="edge_36_save/RestoreV2", _device="/job:localhost/replica:0/task:0/device:CPU:0"](save/RestoreV2:15)]]

What's wrong?

**Comment by MaybeShewill-CV** [comment]
@Spatialghost The repo has been updated you may test it again:)

**Comment by LJXLJXLJX** [comment]
@MaybeShewill-CV 
It works, thank you !
But it seems that you have deleted the Chinese version branch...
I think you may get more stars if you write a complete instruction for Chinese OCR.

**Comment by MaybeShewill-CV** [comment]
@LJXLJXLJX Chinese OCR is almost the same as the synth90k dataset. You just need to prepare your own dataset and train the network:)

**Comment by LJXLJXLJX** [comment]
@MaybeShewill-CV 
Thanks, but I don't know how to establish the char_dict.
There is no entry point in the establish_char_dict.py.
I have already build my char set file, it has one character on each line.
Then how can I use the establish_char_dict.py to generate my char_dict.json and ord_map.json?

**Comment by MaybeShewill-CV** [comment]
@LJXLJXLJX The write tfrecords tools will automatically generate char dict in ./data folder:)

**Comment by LJXLJXLJX** [comment]
@MaybeShewill-CV 
Thank you , I have read your source code carefully, but the organization of my data set maybe a little different from yours. 
I have a folder, there are many images of text in it, and their names are random strings like "20436921_3966525123.jpg". And I have two label files for train and test respectively.
They look like as follow
> 20466593_580910579.jpg 有信心在自己的郑州首
> 20466718_255190113.jpg 市迪马实业股份有限公
> 20466828_2842420474.jpg 海峰在以20比11率
> 20466937_1694775794.jpg 及时提醒查询回复信息
> 20467062_1945108840.jpg 学生应主抓两块，孩子
> 20467140_3726963738.jpg 将消耗队员们大量的体
> 20467296_3193810029.jpg 宜给他们吃纤维太高的
> 20467468_1321113967.jpg ，近日他公开表示自己
> 20467578_2369053802.jpg .cn/bbn7/s
> 20467703_2219947685.jpg 眼折射出广州在历史文
> 20467812_2501136958.jpg 失败的案例却屡见不鲜

I guess these two files correspond to "annotation_train.txt" and "annotation_test.txt" which occurs in your source code.

But I don't know what 'annotation_val.txt' and 'lexicon.txt' means. Could you give me any suggestions, or just improve the README.md for developers who need to train their own model?

Sorry to bother you so much.

**Comment by MaybeShewill-CV** [comment]
@LJXLJXLJX The annotation_train.txt record the image path and the label index. The lexicon.txt record the label. The may find the label via index and the lexicon file. You can refer to the Synth90k dataset website for details:)

**Comment by LJXLJXLJX** [comment]
@MaybeShewill-CV 
Thank you , I have download the Synth90k dataset myself.
I think it's organization is very inefficient, and there are only words, no sentence in fact...
If anyone who wants to train a model to recognize sentences, there are still some works to do.

**Comment by MaybeShewill-CV** [comment]
@LJXLJXLJX No free lunch in the world:)
